### PR TITLE
adding day n° + week number

### PR DESCRIPTION
**Pull request by pierre-kci**

### CODE REVIEW

File: lib/src/gantt_default_week_header.dart [hunk 1]
```diff
@@ -1,5 +1,14 @@
 import 'package:flutter/material.dart';
 
+// TODO extend display with week number
+extension WeeksBetween on DateTime {
+  int get weekNumberOfTheYear => (DateTime.utc(year, month, day)
```
Comment (ahmednfwela) [inline lib/src/gantt_default_week_header.dart:5]: 1. I suggest that instead of an extension method which might confuse users, you can add a function inside `GanttChartDefaultWeekHeader` , e.g. : `int getWeekNumberOfTheYear(DateTime src)`
2. please provide tests for this new method to check on edge cases like
  2.1. first week of the year
  2.2. middle week
  2.3. last week of the year

File: lib/src/gantt_default_day_header.dart [hunk 2]
```diff
@@ -29,7 +29,7 @@ class GanttChartDefaultDayHeader extends StatelessWidget {
       ),
       child: Center(
         child: Text(
-          weekDay.symbol,
+          '${weekDay.symbol}\n${date.day}',
```
Comment (ahmednfwela) [inline lib/src/gantt_default_day_header.dart:32]: wouldn't it make sense to just append it instead of a new line ?
e.g.: 
`SU 13` vs 
```
SU
13
```
this also saves the 10 pixels of added height
AND users can customize this anyway to suit their needs

File: example/lib/main.dart [hunk 3]
```diff
@@ -163,9 +157,9 @@ class _MyHomePageState extends State<MyHomePage> {
                   displayName: '8',
                 ),
                 GanttAbsoluteEvent(
-                  displayName: 'Absoulte Date event',
-                  startDate: DateTime(2022, 6, 7),
-                  endDate: DateTime(2022, 6, 20),
+                  displayName: 'Absolute Date event',
+                  startDate: DateTime.now().subtract(const Duration(days: 7)),
```
Comment (ahmednfwela) [inline example/lib/main.dart:161]: how should the gantt chart display an event that starts before the chart's start date ?

File: example/lib/main.dart [hunk 4]
```diff
@@ -104,18 +98,18 @@ class _MyHomePageState extends State<MyHomePage> {
...
               maxDuration: const Duration(days: 30 * 2),
-              startDate: DateTime(2022, 6, 7),
+              startDate: DateTime.now(),
               dayWidth: dayWidth,
               eventHeight: 40,
               stickyAreaWidth: 200,
               showStickyArea: showStickyArea,
               showDays: showDaysRow,
-              weekEnds: const {WeekDay.friday, WeekDay.saturday},
+              weekEnds: const {WeekDay.saturday, WeekDay.sunday},
```
Comment (ahmednfwela) [inline example/lib/main.dart:107]: please revert `weekEnds` and `startOfTheWeek` back, since in this example I am showing the default values

File: lib/src/week_day.dart [hunk 5]
```diff
@@ -1,3 +1,5 @@
+// Consider localizing this
```
Comment (ahmednfwela) [inline lib/src/week_day.dart:1]: localization can be done on user side, since this will add unwanted overhead to this package

File: lib/src/gantt_default_week_header.dart [hunk 6]
```diff
@@ -27,7 +36,8 @@ class GanttChartDefaultWeekHeader extends StatelessWidget {
           } else if (constraints.maxWidth < 7 * 20) {
             txt = '${weekDate.month}-${weekDate.year % 100}';
           } else {
-            txt = '${weekDate.day}-${weekDate.month}-${weekDate.year}';
+            txt =
+                '${weekDate.day}-${weekDate.month}-${weekDate.year}  #${weekDate.weekNumberOfTheYear} ';
```
Comment (ahmednfwela) [inline lib/src/gantt_default_week_header.dart:40]: please remove the extra space at the end

**Review by ahmednfwela — CHANGES_REQUESTED** [review]
Thanks for this great PR, please address the review comments, and change the screenshot to the current state of the example in  `/doc/demo.png`

File: lib/src/gantt_default_day_header.dart (hunk 2 above)
Comment (pierre-kci) [reply lib/src/gantt_default_day_header.dart:32]: image

If I append then sometimes space isn't sufficient and UI automatically add a new line adding incoherence
It sure might be possible to dig this further to that fonts are resized accordingly, but I do not master this,

Also, it looks clean to read days number on a horizontal line,
But this is up to you

File: example/lib/main.dart (hunk 3 above)
Comment (pierre-kci) [reply example/lib/main.dart:161]: true

File: example/lib/main.dart (hunk 4 above)
Comment (pierre-kci) [reply example/lib/main.dart:107]: Super interesting, I suspected that Egypt only did week-ends on Friday Saturday,
Most of the world do Saturday Sunday, or Sunday only. 
It's cool to keep Egyptian vibes and through this to curb ethnocentrism a bit.

wekeendmap-eng

File: lib/src/gantt_default_week_header.dart (hunk 1 above)
Comment (pierre-kci) [reply lib/src/gantt_default_week_header.dart:5]: well spotted, 1st week of the year required evol

File: example/lib/main.dart (hunk 4 above)
Comment (ahmednfwela) [reply example/lib/main.dart:107]: well, it's ok if you want to change the default weekend values, what I meant is that you need to also change the readme as well

my way of thinking is that if a user can change the behavior anyway, the defaults are irrelevant

File: lib/src/gantt_default_day_header.dart (hunk 2 above)
Comment (ahmednfwela) [reply lib/src/gantt_default_day_header.dart:32]: well, you have a valid point, I think it's better if the number is on a new line

**Comment by pierre-kci** [comment]
ok, it seems all points are addressed.
Do you see any changes still pending ?


Le mar. 10 janv. 2023 à 18:54, Ahmed Fwela ***@***.***> a
écrit :

**Comment by ahmednfwela** [comment]
I can only see the initial commit in this 0R,did you make any edits?

**Comment by pierre-kci** [comment]
ok, git mystery, I opened this other PR : https://github.com/Bdaya-Dev/flutter_gantt_chart/pull/6
